core/runtime/execute_engine.cpp

```diff
@@ -58,6 +59,29 @@ RTDevice select_rt_device(const RTDevice& engine_device, const RTDevice& curr_de
   return new_target_device_opt.value();
 }
 
+bool _cudagraphs_validate_shapes(std::vector<at::Tensor> inputs, c10::intrusive_ptr<TRTEngine> compiled_engine) {
+  // Validate whether the current input shapes to the engine
+  // invalidate the existing cudagraphs object
+
+  // Populate the shape key for the inputs
+  std::stringstream new_shape_key_ss;
+  for (auto input : inputs) {
+    new_shape_key_ss << input.sizes();
```

Comment:
Does this create the same shape key as python?

Reply:
I believe so - I will check the bracketing to see if `.sizes()` uses `[]` or `()`, but the scheme is the same. I will remove the whitespace in this case as well.